chainntnfs/btcdnotify/btcd.go

```diff
@@ -613,27 +613,58 @@ func (b *BtcdNotifier) confDetailsManually(txid *chainhash.Hash,
 	return nil, nil
 }
 
-// handleBlocksConnected applies a chain update for a new block. Any watched
+// handleBlockConnected applies a chain update for a new block. Any watched
 // transactions included this block will processed to either send notifications
 // now or after numConfirmations confs.
 // TODO(halseth): this is reusing the neutrino notifier implementation, unify
 // them.
-func (b *BtcdNotifier) handleBlockConnected(newBlock *filteredBlock) error {
-	// First we'll notify any subscribed clients of the block.
+func (b *BtcdNotifier) handleBlockConnected(epoch chainntnfs.BlockEpoch) error {
+	// First process the block for our internal state. A new block has
+	// been connected to the main chain. Send out any N confirmation
+	// notifications which may have been triggered by this new block.
+	rawBlock, err := b.chainConn.GetBlock(epoch.Hash)
```

Comment:
Why are we fetching the block again? We already fetch the raw block here: https://github.com/lightningnetwork/lnd/blob/6d62f12eb19808aac6f312240ea4c9995f65919d/chainntnfs/btcdnotify/btcd.go#L385

Reply:
I believe it saves a bit of code duplication since otherwise we'd have to do the same filtered block construction here: https://github.com/lightningnetwork/lnd/pull/1439/files#diff-0bfcf75d8a29f4e9d616031ff3e5a805R400